packages/xo-web/src/common/xo/index.js

```diff
@@ -3796,3 +3798,29 @@ const _callGithubApi = async (endpoint = '') => {
 export const getMasterCommit = () => _callGithubApi('/commits/master')
 
 export const compareCommits = (base, head) => _callGithubApi(`/compare/${base}...${head}`)
+
+export const uuidToLink = text => {
+  const regex = /\b[0-9a-f]{8}\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\b[0-9a-f]{12}\b/g
+  const parts = text.split(regex)
+  const uuids = text.match(regex) || []
+  const state = store.getState()
+
+  return flatMap(parts, (part, index) => {
+    const uuid = uuids[index]
+    if (uuid) {
+      const object = getObject(state, uuid)
+      if (object && ['pool', 'host', 'VM', 'SR'].includes(object.type)) {
```

Comment:
Not necessarily for this PR but maybe we can also match XO-specific objects like users and link to something like: `/settings/users?s=uuid:<UUID>`.